lib/ts/BufferWriter.h

```diff
@@ -708,6 +711,31 @@ operator<<(BufferWriter &w, int const &i)
   return bwformat(w, BWFSpec::DEFAULT, static_cast<intmax_t>(i));
 }
 
+//-- Floating type(s)
+inline BufferWriter &
+bwformat(BufferWriter &w, BWFSpec const &spec, double const &d)
+{
+  return d < 0 ? bw_fmt::Format_Floating(w, spec, -d, true) : bw_fmt::Format_Floating(w, spec, d, false);
+}
+
+inline BufferWriter &
+bwformat(BufferWriter &w, BWFSpec const &spec, float const &f)
+{
+  return f < 0 ? bw_fmt::Format_Floating(w, spec, -f, true) : bw_fmt::Format_Floating(w, spec, f, false);
+}
+
+inline BufferWriter &
```

Comment:
These cases should be handled by a generic template, it shouldn't be necessary to write them explicitly.